### PR TITLE
Fix backward of `F.average_pooling_nd` with pad_value of None

### LINKED ISSUE #6210: `AveragePoolingNDGrad._get_pooling_width` is not implemented
In `AveragePoolingNDGrad`, `self._get_pooling_width` is used (I guess it should be `self.apoolnd._get_pooling_width` instead) when `self.pad_value` is None while
the class does not define the method and the tests of `F.average_pooling_nd` succeed.

Current `TestAveragePoolingND` does parameterize `pad_value` but does not use it in forward computation for backward tests. The default value of pad_value in F.average_pooling_nd is 0.

https://github.com/chainer/chainer/blob/d7b9bfd0e77fd79d58c4ae04cc6a1ac37ce66e8a/chainer/functions/pooling/average_pooling_nd.py#L184-L185
https://github.com/chainer/chainer/blob/d7b9bfd0e77fd79d58c4ae04cc6a1ac37ce66e8a/chainer/functions/pooling/average_pooling_nd.py#L201-L202

https://github.com/chainer/chainer/blob/d7b9bfd0e77fd79d58c4ae04cc6a1ac37ce66e8a/tests/chainer_tests/functions_tests/pooling_tests/test_average_pooling_nd.py#L20-L24
https://github.com/chainer/chainer/blob/d7b9bfd0e77fd79d58c4ae04cc6a1ac37ce66e8a/tests/chainer_tests/functions_tests/pooling_tests/test_average_pooling_nd.py#L140-L148

---

(2019/02/13): Backward tests fail if I just copy `_get_pooling_width` from `AveragePoolingND`.

### PR DESCRIPTION
**Pull request by crcrpar**
Closes #6210 and related to #6071 

- [x] test_backward of AveragePoolingNDGrad.forward_cpu
- [x] test_backward of AveragePoolingNDGrad.forward_gpu
- [x] test_double_backward
- [x] consistency regression to `F.average_pooling_2d` forward
- [x] consistency regression to `F.average_pooling_2d` backward (numpy and cupy imple.)

### CODE REVIEW

**Comment by crcrpar** [comment]
Travis CI failures seem unrelated to this PR.

**Comment by takagi** [comment]
I've restarted the Travis jobs that happened to fail.

**Comment by takagi** [comment]
Jenkins, test this please.

**Comment by takagi** [comment]
Jenkins, test this please.

File: chainer/functions/pooling/average_pooling_nd.py [hunk 1]
```diff
@@ -177,14 +180,15 @@ def forward_cpu(self, gys):
         idims = self._in_shape[2:]
         odims = gy.shape[2:]
         colon = slice(None, None, None)
+        is_pad_value_none = self.pad_value is None
+        if is_pad_value_none:
```
[the remark this answers is not in the document]
Comment (crcrpar) [reply chainer/functions/pooling/average_pooling_nd.py]: Reasons
- The shape of `width`(`coeff`) in `AveragePoolingNDGrad.forward` differs from that in `AveragePooling.forward`.
- Dividing `gy` by `width`(`coeff`) when `pad_value is None` is the same as the implementation of ChainerX which is tested. 
    - ref: https://github.com/chainer/chainer/blob/a8f9cb5500def0037c0b68f2ad8901dbca52bdb1/chainerx_cc/chainerx/native/native_device/pool.cc#L253-L257

File: chainer/functions/pooling/average_pooling_nd.py [hunk 2]
```diff
@@ -193,31 +197,36 @@ def forward_gpu(self, gys):
             x, = self.apoolnd.get_retained_inputs()
             return self.apoolnd.backward_gpu((x.data,), gys)
 
+        is_pad_value_none = self.pad_value is None
+
         gy, = gys
         n, c = self._in_shape[:2]
         idims = self._in_shape[2:]
         odims = gy.shape[2:]
+        if is_pad_value_none:
```
[the remark this answers is not in the document]
Comment (crcrpar) [reply chainer/functions/pooling/average_pooling_nd.py]: Ditto.

File: chainer/functions/pooling/average_pooling_nd.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (takagi) [reply chainer/functions/pooling/average_pooling_nd.py]: Thanks. That makes sense!

File: chainer/functions/pooling/average_pooling_nd.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (takagi) [reply chainer/functions/pooling/average_pooling_nd.py]: Thanks, too!

File: chainer/functions/pooling/average_pooling_nd.py [hunk 3]
```diff
@@ -177,14 +180,15 @@ def forward_cpu(self, gys):
         idims = self._in_shape[2:]
         odims = gy.shape[2:]
         colon = slice(None, None, None)
+        is_pad_value_none = self.pad_value is None
+        if is_pad_value_none:
+            width = getattr(self.apoolnd, 'width')
```
Comment (takagi) [inline chainer/functions/pooling/average_pooling_nd.py]:
```suggestion
            width = self.apoolnd.width
```

File: chainer/functions/pooling/average_pooling_nd.py [hunk 4]
```diff
@@ -193,31 +197,36 @@ def forward_gpu(self, gys):
...
             return self.apoolnd.backward_gpu((x.data,), gys)
 
+        is_pad_value_none = self.pad_value is None
+
         gy, = gys
         n, c = self._in_shape[:2]
         idims = self._in_shape[2:]
         odims = gy.shape[2:]
+        if is_pad_value_none:
+            coeff = getattr(self.apoolnd, 'coeff')
```
Comment (takagi) [inline chainer/functions/pooling/average_pooling_nd.py]:
```suggestion
            coeff = self.apoolnd.coeff
```

**Comment by takagi** [comment]
Jenkins, test this please.

File: chainer/functions/pooling/average_pooling_nd.py (hunk 3 above)
Comment (crcrpar) [reply chainer/functions/pooling/average_pooling_nd.py]: Thank you.

**Comment by takagi** [comment]
LGTM! I'm going to merge this. The Jenkins CI failure above is not related to this PR, which is because of  https://github.com/chainer/chainer/pull/6393.

**Pull request merged**